### ISSUE TITLE
Installation on Raspberry Pi 4 causes homebridge.service to fail with result 'exit code'. 

### ISSUE THREAD

**Issue by idealdreams** [nzapponi/homebridge-simplisafe3#51]
First, this is an amazing plugin. Thank you for the work done on this. I am currently using this plugin successfully on Homebridge running on my MacBook. I purchased a Raspberry Pi as I want to move Homebridge from my Mac to a standalone "server." 

I have installed all plugins successfully except for this plugin. When I install it, update the config file, and restart Homebridge, it gets stuck in a reboot loop until I stop the process. When I look at the logs, I see "homebridge.service: failed with result 'exit code'". 

I uninstalled and tried installing with `npm install -g --unsafe-perm homebridge-simplisafe3` to the same issues. The below errors appeared after the install:

```
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @ffmpeg-installer/darwin-x64@4.1.0 (node_modules/homebridge-simplisafe3/node_modules/@ffmpeg-installer/darwin-x64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @ffmpeg-installer/darwin-x64@4.1.0: wanted {"os":"darwin","arch":"x64"} (current: {"os":"linux","arch":"arm"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @ffmpeg-installer/linux-arm64@4.1.4 (node_modules/homebridge-simplisafe3/node_modules/@ffmpeg-installer/linux-arm64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @ffmpeg-installer/linux-arm64@4.1.4: wanted {"os":"linux","arch":"arm64"} (current: {"os":"linux","arch":"arm"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @ffmpeg-installer/linux-ia32@4.1.0 (node_modules/homebridge-simplisafe3/node_modules/@ffmpeg-installer/linux-ia32):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @ffmpeg-installer/linux-ia32@4.1.0: wanted {"os":"linux","arch":"ia32"} (current: {"os":"linux","arch":"arm"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @ffmpeg-installer/linux-x64@4.1.0 (node_modules/homebridge-simplisafe3/node_modules/@ffmpeg-installer/linux-x64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @ffmpeg-installer/linux-x64@4.1.0: wanted {"os":"linux","arch":"x64"} (current: {"os":"linux","arch":"arm"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @ffmpeg-installer/win32-ia32@4.1.0 (node_modules/homebridge-simplisafe3/node_modules/@ffmpeg-installer/win32-ia32):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @ffmpeg-installer/win32-ia32@4.1.0: wanted {"os":"win32","arch":"ia32"} (current: {"os":"linux","arch":"arm"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @ffmpeg-installer/win32-x64@4.1.0 (node_modules/homebridge-simplisafe3/node_modules/@ffmpeg-installer/win32-x64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @ffmpeg-installer/win32-x64@4.1.0: wanted {"os":"win32","arch":"x64"} (current: {"os":"linux","arch":"arm"})
```

I am not sure how to proceed from here. What am I missing?

**Comment by nikonratm** [comment]
These kind of problems are often to do with a config file or an incorrect version of software (eg node). The log lines above are just a warning about ffmpeg which is not required so that shouldn’t be the problem. I’m assuming there aren’t any other lines in the log that might help? Did you try resetting your config file?

**Comment by idealdreams** [comment]
I appreciate the response. 

The Raspberry Pi is brand new and is running the latest version of Homebridge and Node (or at least I would assume the latest versions would be installed when I fresh installed yesterday?). I don't think it's the config file as it works fine before I install. I tried updating the config to include the parameters for the plugin and tried it without as well to no avail.

Here's additional lines of the log, not sure if there's anything more in here that would help. 

`-- Logs begin at Sat 2019-12-28 07:17:01 HST. --
Dec 28 07:23:33 raspberrypi homebridge[666]: [12/28/2019, 7:23:33 AM] Loaded plugin: homebridge-nest
Dec 28 07:23:33 raspberrypi homebridge[666]: [12/28/2019, 7:23:33 AM] Registering platform 'homebridge-nest.Nest'
Dec 28 07:23:33 raspberrypi homebridge[666]: [12/28/2019, 7:23:33 AM] ---
Dec 28 07:23:33 raspberrypi homebridge[666]: [12/28/2019, 7:23:33 AM] Loaded plugin: homebridge-webos-tv
Dec 28 07:23:33 raspberrypi homebridge[666]: [12/28/2019, 7:23:33 AM] Registering accessory 'homebridge-webos-tv.webostv'
Dec 28 07:23:33 raspberrypi homebridge[666]: [12/28/2019, 7:23:33 AM] ---
Dec 28 07:23:33 raspberrypi homebridge[666]: [12/28/2019, 7:23:33 AM] Loading 2 platforms...
Dec 28 07:23:33 raspberrypi homebridge[666]: [12/28/2019, 7:23:33 AM] [Nest] Initializing Nest platform...
Dec 28 07:23:33 raspberrypi homebridge[666]: [12/28/2019, 7:23:33 AM] [Nest] Fetching Nest devices.
Dec 28 07:23:33 raspberrypi homebridge[666]: [12/28/2019, 7:23:33 AM] [Config] Initializing config platform...
Dec 28 07:23:33 raspberrypi homebridge[666]: [12/28/2019, 7:23:33 AM] [Config] Spawning homebridge-config-ui-x with PID 688
Dec 28 07:23:33 raspberrypi homebridge[666]: [12/28/2019, 7:23:33 AM] Loading 3 accessories...
Dec 28 07:23:33 raspberrypi homebridge[666]: [12/28/2019, 7:23:33 AM] [Garage Door] Initializing Chamberlain accessory...
Dec 28 07:23:33 raspberrypi homebridge[666]: [12/28/2019, 7:23:33 AM] [LG OLED E8] Initializing webostv accessory...
Dec 28 07:23:33 raspberrypi homebridge[666]: [12/28/2019, 7:23:33 AM] [LG LED TV] Initializing webostv accessory...
Dec 28 07:23:33 raspberrypi homebridge[666]: Failed to find plugin to handle accessory SimpliSafe 3
Dec 28 07:23:33 raspberrypi homebridge[666]: Failed to find plugin to handle accessory Front Door
Dec 28 07:23:33 raspberrypi homebridge[666]: Failed to find plugin to handle accessory Back Door
Dec 28 07:23:33 raspberrypi homebridge[666]: Failed to find plugin to handle accessory Basement
Dec 28 07:23:33 raspberrypi homebridge[666]: Failed to find plugin to handle accessory Garage
Dec 28 07:23:33 raspberrypi homebridge[666]: Failed to find plugin to handle accessory Side Door
Dec 28 07:23:33 raspberrypi homebridge[666]: [12/28/2019, 7:23:33 AM] [LG OLED E8] webOS - connected to TV
Dec 28 07:23:33 raspberrypi homebridge[666]: [12/28/2019, 7:23:33 AM] [LG OLED E8] webOS - app launched, current appId: com.webos.app.hdmi1
Dec 28 07:23:34 raspberrypi homebridge[666]: [12/28/2019, 7:23:34 AM] [LG OLED E8] webOS - audio status changed
Dec 28 07:23:34 raspberrypi homebridge[666]: [12/28/2019, 7:23:34 AM] [LG OLED E8] webOS - current volume: 11
Dec 28 07:23:34 raspberrypi homebridge[666]: [12/28/2019, 7:23:34 AM] [LG OLED E8] webOS - muted: No
Dec 28 07:23:34 raspberrypi homebridge[666]: [12/28/2019, 7:23:34 AM] [LG OLED E8] webOS - sound output changed
Dec 28 07:23:34 raspberrypi homebridge[666]: [12/28/2019, 7:23:34 AM] [LG OLED E8] webOS - current sound output: external_arc
Dec 28 07:23:34 raspberrypi homebridge[666]: [12/28/2019, 7:23:34 AM] [LG OLED E8] webOS - current channel status changed
Dec 28 07:23:34 raspberrypi homebridge[666]: [12/28/2019, 7:23:34 AM] [LG OLED E8] webOS - current channel: undefined, undefined
Dec 28 07:23:35 raspberrypi homebridge[666]: [12/28/2019, 7:23:35 AM] [Nest] initing thermostat "Family Room": deviceId: XXX structureId: XXX
Dec 28 07:23:35 raspberrypi homebridge[666]: [12/28/2019, 7:23:35 AM] [Nest] Initializing platform accessory 'Family Room'...
Dec 28 07:23:35 raspberrypi homebridge[666]: Setup Payload:
Dec 28 07:23:35 raspberrypi homebridge[666]: X-HM://0024CBDCDA2EP
Dec 28 07:23:35 raspberrypi homebridge[666]: [12/28/2019, 7:23:35 AM] Homebridge is running on port XXX.
Dec 28 07:23:36 raspberrypi homebridge[666]: [12/28/2019, 7:23:36 AM] [Config] Homebridge Config UI X v4.6.7 is listening on :: port XXX
Dec 28 07:23:36 raspberrypi homebridge[666]: [12/28/2019, 7:23:36 AM] [LG LED TV] webOS - Error: connect EHOSTUNREACH XXX.XXX.XX.XXX:XXXX
Dec 28 07:23:36 raspberrypi homebridge[666]:     at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1129:14) {
Dec 28 07:23:36 raspberrypi homebridge[666]:   errno: 'EHOSTUNREACH',
Dec 28 07:23:36 raspberrypi homebridge[666]:   code: 'EHOSTUNREACH',
Dec 28 07:23:36 raspberrypi homebridge[666]:   syscall: 'connect',
Dec 28 07:23:36 raspberrypi homebridge[666]:   address: ‘XXX.XXX.XX.XXX’,
Dec 28 07:23:36 raspberrypi homebridge[666]:   port: XXXX
Dec 28 07:23:36 raspberrypi homebridge[666]: }
Dec 28 07:23:43 raspberrypi homebridge[666]: [12/28/2019, 7:23:43 AM] [Config] [HapClient] [XXX.XX.XX.2:62043 (DF:0C:7D:81:EF:65)] 401 Unauthorized - make sure Homebridge pin for this instance is set to XXX-XX-XXX.
Dec 28 07:54:12 raspberrypi homebridge[666]: [12/28/2019, 7:54:12 AM] [Config] [homebridge] Error: ETIMEDOUT
Dec 28 07:56:08 raspberrypi homebridge[666]: [12/28/2019, 7:56:08 AM] [Config] [homebridge-nest] Error: ETIMEDOUT
Dec 28 09:01:39 raspberrypi homebridge[666]: [12/28/2019, 9:01:39 AM] Got SIGTERM, shutting down Homebridge...
Dec 28 09:01:39 raspberrypi systemd[1]: Stopping Homebridge...
Dec 28 09:01:44 raspberrypi systemd[1]: homebridge.service: Main process exited, code=exited, status=143/n/a
Dec 28 09:01:44 raspberrypi systemd[1]: homebridge.service: Failed with result 'exit-code'.
Dec 28 09:01:44 raspberrypi systemd[1]: Stopped Homebridge.
Dec 28 09:01:44 raspberrypi systemd[1]: homebridge.service: Found left-over process 688 (homebridge-conf) in control group while starting unit. Ignoring.
Dec 28 09:01:44 raspberrypi systemd[1]: This usually indicates unclean termination of a previous run, or service implementation deficiencies.
Dec 28 09:01:44 raspberrypi systemd[1]: Started Homebridge.
`

**Comment by nikonratm** [comment]
Depending on how you installed Homebridge and node they may not be the correct versions, FWIW.

The logs show a few things, one if which is that this plugin isnt starting properly. When you ported things over from your Mac did you move all of the Homebridge files, i.e. `.homebridge/persist` and `.homebridge/accessories` directories? Moving just the config may not be enough.

But also your logs show a problem with the webOS plugin and the homebridge-nest plugin:
`
Dec 28 07:23:36 raspberrypi homebridge[666]: [12/28/2019, 7:23:36 AM] [LG LED TV] webOS - Error: connect EHOSTUNREACH XXX.XXX.XX.XXX:XXXX
Dec 28 07:23:36 raspberrypi homebridge[666]:     at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1129:14) {
Dec 28 07:23:36 raspberrypi homebridge[666]:   errno: 'EHOSTUNREACH',
Dec 28 07:23:36 raspberrypi homebridge[666]:   code: 'EHOSTUNREACH',
Dec 28 07:23:36 raspberrypi homebridge[666]:   syscall: 'connect',
Dec 28 07:23:36 raspberrypi homebridge[666]:   address: ‘XXX.XXX.XX.XXX’,
Dec 28 07:23:36 raspberrypi homebridge[666]:   port: XXXX
Dec 28 07:23:36 raspberrypi homebridge[666]: }
Dec 28 07:23:43 raspberrypi homebridge[666]: [12/28/2019, 7:23:43 AM] [Config] [HapClient] [XXX.XX.XX.2:62043 (DF:0C:7D:81:EF:65)] 401 Unauthorized - make sure Homebridge pin for this instance is set to XXX-XX-XXX.
Dec 28 07:54:12 raspberrypi homebridge[666]: [12/28/2019, 7:54:12 AM] [Config] [homebridge] Error: ETIMEDOUT
Dec 28 07:56:08 raspberrypi homebridge[666]: [12/28/2019, 7:56:08 AM] [Config] [homebridge-nest] Error: ETIMEDOUT
`

Those are both networking (or perhaps password) issues, is your Pi connected to the internet properly?

I'd suggest this: https://github.com/nfarina/homebridge/wiki/Basic-Troubleshooting
And trying each of your plugins one-by-one, not just to see if Homebridge crashes but also make sure they each work, I suspect they wont.

Anyway, unlikely an issue with this plugin but more to do with your Homebridge / RPi setup in general so perhaps posting to forums / reddit more specifically for that, as opposed to here. Also you may want to explore:
- https://www.github.com/oznu/homebridge-config-ui-x
- OR even http://hoobs.org

Please close this issue when appropriate as again, I dont think its necessarily anything to do with this plugin.

**Comment by idealdreams** [comment]
Closing out this issue. I had another instance of Homebridge running on my network (on my MacBook) while I was configuring the instance on the Raspberry Pi. It must have crossed wires trying to connect to the same service on the same network.

I uninstalled Homebridge on my MacBook, wiped all traces clean from the system files, rebooted, and then reinstalled to the Raspberry Pi. No more errors and it can now successfully restart. 

No more issues. I've now ported everything over to the Raspberry Pi and am running solely off of that. 

Thanks for the assistance!
